controllers/openshift_controller.go

```diff
@@ -1805,6 +1805,55 @@ func (r *KataConfigOpenShiftReconciler) processDoneNode(node *corev1.Node) error
 	return nil
 }
 
+// If multiple errors occur during execution of this function the last one
+// will be returned.
+func (r *KataConfigOpenShiftReconciler) updateStatusNew() error {
+
+	if r.kataConfig.Status.UnInstallationStatus.InProgress.IsInProgress != corev1.ConditionTrue &&
+		r.kataConfig.Status.InstallationStatus.IsInProgress != corev1.ConditionTrue {
+		return nil
+	}
+
+	err, nodeList := r.getNodes()
+	if err != nil {
+		return err
+	}
+
+	r.clearInstallStatus()
+	r.clearUninstallStatus()
+
+	r.kataConfig.Status.TotalNodesCount = func() int {
+		err, nodes := r.getNodesWithLabels(r.getNodeSelectorAsMap())
+		if err != nil {
+			r.Log.Info("Error retrieving kata-oc labelled Nodes to count them", "err", err)
+			return 0
+		}
+		return len(nodes.Items)
+	}()
+
+	for _, node := range nodeList.Items {
+		if annotation, ok := node.Annotations["machineconfiguration.openshift.io/state"]; ok {
+			switch annotation {
+			case "Done":
```

Comment:
I see there are constants for the state annotation in the MCO source tree. I'm wondering if they should be used instead of hardcoded strings ?

Reply:
That's a good idea, I can't find them though, can you point me to them?  Are they public?

I only used literals because the legacy implementation used them and frankly it didn't occur to me that they might be defined somewhere in the MCO. :-)

Reply:
These constants are defined at the machine config daemon level. They are accessible but I don't know if they are actually public.

States are defined here :

https://github.com/openshift/machine-config-operator/blob/master/pkg/daemon/constants/constants.go#L36-L44

Note there's also an `"Unreconcilable"` state as well. Not sure if it needs to be handled as well.

The annotation key is also defined BTW :

https://github.com/openshift/machine-config-operator/blob/master/pkg/daemon/constants/constants.go#L16

Reply:
Thanks, I'm working to fix this.